platforms/documentation/docs/src/snippets/failures/verificationFailure/groovy/build.gradle

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright 2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// tag::verification-failure[]
+def logFile = layout.buildDirectory.file("build/processed.log").get().asFile
+
+tasks.register("process") {
+    outputs.files(logFile) // <1>
+
+    doLast {
+        logFile << "Step 1 Complete." // <2>
+        throw new VerificationException("Process failed!") // <3>
+        logFile << "Step 2 Complete." // <4>
+    }
```

Comment:
❌ There are a couple of problems with this....

1. The output path is now `build/build/processed.log`
2. The call to `get().asFile` is just as bad as hardcoding the path to build.  We can only resolve the provider until the task's action.

```suggestion
tasks.register("process") {
    def outputFile = layout.buildDirectory.file("processed.log")
    outputs.files(outputFile) // <1>

    doLast {
        def logFile = outputFile.get().asFile
        logFile << "Step 1 Complete." // <2>
        throw new VerificationException("Process failed!") // <3>
        logFile << "Step 2 Complete." // <4>
    }
```

Reply:
Got it, resolved.